Hot fix: update stats v4 endpoint and banner message when reverting from stats v4 to v3

jaclync opened the pull request:
Fixes #1489 

## Changes
- Created a new Woo API version case `wcAnalytics = "wc-analytics"` for the new API namespace that replaces `mark4 = "wc/v4"` for stats in WC Admin plugin v0.22+
- Used the new WC analytics API namespace for the stats v4 endpoint
- In the banner where we show when the user was previously seeing stats v4 UI and now is not eligible for stats v4 (with the new namespace), updated the messaging to mention the minimum WC Admin plugin version (0.22) for new stats UI

## Testing

### Test case 1: the store has WC Admin v0.22+

- In `wp-admin`, make sure the WC Admin plugin is v0.22+ and activated
- Launch the app from the PR branch --> should see the new stats, or the banner to view the new stats
- In `wp-admin`, deactivate the WC Admin plugin
- Back in the app, pull down to refresh or relaunch the app --> should see the stats reverted to v3 UI, and a banner suggesting updating the WC Admin plugin to v0.22+ as in the screenshot

### Test case 2: the store has WC Admin with a version lower than 0.22

- In `wp-admin`, make sure the WC Admin plugin is lower than v0.22 and activated (you can download an older WC Admin version from the [WC Admin GitHub releases page](https://github.com/woocommerce/woocommerce-admin/releases?after=v0.20.1))
- Launch the app from `develop` and opt-in the new stats if it's not currently shown
- Launch the app from the PR branch --> should see the stats reverted to v3 UI, and a banner suggesting updating the WC Admin plugin to v0.22+ as in the screenshot
- In `wp-admin`, update the WC Admin plugin to v0.22+ and make sure it's activated
- Launch the app from the PR branch --> should see the banner to view the new stats
- Tap "Try It Now" on the banner to view the new stats --> should see the new stats UI



## Example screenshots

When the user used to see stats v4 UI but isn't eligible for stats v4 anymore (either doesn't have WC Admin installed/activated or the WC Admin version is below 0.22), we show this banner:

<img src="https://user-images.githubusercontent.com/1945542/69119232-e6361200-0ad0-11ea-968b-f20aef79f2f5.png" width="350" />



Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

jaclync committed:
Use the new WC Analytics API namespace, and update the banner message when we revert from v4 to v3.

jaclync committed:
Move the string to a new line to fix line length warning.

jaclync committed:
Add an entry to the release notes.

jaclync commented:
@pmusolino @mindgraffiti could one of you take a look at this hotfix PR for release 3.1 when you get a chance today? More details in #1489. cc @JavonDavis 

As an overview for Paolo, we added a new stats UI for sites with WooCommerce Admin plugin activated. Here's a screenshot comparison btw v3 (old) and v4 (new) stats UI, I usually tell by the tab labels ("Days"/"Weeks"/"Months"/"Years" vs. "Today"/"This week"/"This month"/"This year"):

v3 stats UI | v4 stats UI
-- | --
![Simulator Screen Shot - iPhone 11 - 2019-11-19 at 14 13 27](https://user-images.githubusercontent.com/1945542/69121243-cb669c00-0ad6-11ea-9571-5dce6cdb2147.png) | ![Simulator Screen Shot - iPhone 11 - 2019-11-19 at 14 12 12](https://user-images.githubusercontent.com/1945542/69121209-b853cc00-0ad6-11ea-9dc5-36d00302b51d.png)

pmusolino reviewed (approved):
The code looks good to me! ✅ 

I do some tests, following your instructions.
Everything looks good, but the first screen (from develop branch, and on a new installation of the app) doesn't allow me to show the new stats. `Launch the app from develop and opt-in the new stats if it's not currently shown`. There is no opt-in banner. It's normal? 

| Develop branch, no opt-in message             |  After moving to the branch |
:-------------------------:|:-------------------------:
![app develop no opt-in](https://user-images.githubusercontent.com/495617/69136238-cac00c80-0aba-11ea-98a7-5aa5df4c32b5.png) | ![branch](https://user-images.githubusercontent.com/495617/69136293-e5928100-0aba-11ea-9636-835e040891bb.png)

| Branch after opt-in             |  Branch with plugin deactivated |
:-------------------------:|:-------------------------:
![branch opt-in](https://user-images.githubusercontent.com/495617/69136350-00fd8c00-0abb-11ea-80d2-e19671c7eb62.png) | ![branch, plugin deactivated](https://user-images.githubusercontent.com/495617/69136377-09ee5d80-0abb-11ea-9076-29dc162df4b6.png)

mindgraffiti reviewed (commented):
I can't get the new stats to take. I installed the old wc-admin (v.0.21) on my test site, then ran a fresh build on develop. Logged in and activated the new stats. Stopped the build, switched to the hotfix branch, and ran it. I saw this, which was good:

<img src="https://user-images.githubusercontent.com/1062444/69187120-9448b880-0adf-11ea-9cb6-e34aee6ddcff.png" width="350" />

I went back to my website, updated the plugin to 0.22. Returned to the app and tried to pull-to-refresh. Old stats remained. I logged out of the app and logged back in. Old stats remained. I also don't have an option to opt-in to the new stats in Settings:

<img src="https://user-images.githubusercontent.com/1062444/69187219-c8bc7480-0adf-11ea-905e-fa2567738c2f.png" width="350" />

jaclync commented:
> Launch the app from develop and opt-in the new stats if it's not currently shown. There is no opt-in banner. It's normal?

@pmusolino sorry I missed a condition in the test case - in order to see the new stats on `develop` the WC Admin plugin has to be lower than v0.22. I've updated the test cases (with a link to get the older versions of WC Admin plugin in the test case 2), could you try again? 😄

jaclync commented:
> I went back to my website, updated the plugin to 0.22. Returned to the app and tried to pull-to-refresh. Old stats remained. I logged out of the app and logged back in. Old stats remained. I also don't have an option to opt-in to the new stats in Settings:

@mindgraffiti could you check on wp-admin that the WC Admin plugin is activated after the update (try refreshing the Plugins page)? If you don't see the option to opt-in to the new stats in Settings, that means we cannot get any data from the new stats endpoint so the new stats UI is not available. On my site, after I updated the plugin, somehow it got deactivated by default. After reactivating the plugin, I got to see the new stats UI again on the PR branch. See if this is the case for you, and if not, would it be possible to add me as a user to your test store?

mindgraffiti commented:
<img width="882" alt="Screen Shot 2019-11-20 at 8 03 23 AM" src="https://user-images.githubusercontent.com/1062444/69245905-58f4cb00-0b6d-11ea-9b69-ed18324af4b4.png">

@jaclync, yep it deactivated from this error ☝️ and I reactivated it. I tried switching away from the store, coming back to it, using pull-to-refresh, and logging out and logging back in again. It doesn't usually take that long for the reactivated plugin to propagate the new API endpoints. Sending you an admin invite to my test site.

mindgraffiti commented:
I dug a little deeper into this. It looks like the StatsVersionStateCoordinator doesn't take `wcAnalytics` into account, only stats .v4 and .v4. I also logged a few things and noticed that the v4 endpoint was getting requested, despite the the wcAnalytics namespace defined in the remote request.

JavonDavis commented:
Just noting that even though https://github.com/woocommerce/woocommerce-android/pull/1575 is merged, I'm holding off on a new beta until this one is merged as well

pmusolino commented:
> @pmusolino sorry I missed a condition in the test case - in order to see the new stats on develop the WC Admin plugin has to be lower than v0.22. I've updated the test cases (with a link to get the older versions of WC Admin plugin in the test case 2), could you try again? 😄

Now it worked 😉 ✅

jaclync commented:
Hey @mindgraffiti I was able to see the new stats banner and then the UI in your store, though the Order stats all seem to be empty and consistent on wp-admin so it seems to be a separate issue.

![Simulator Screen Shot - iPhone 8 - 2019-11-21 at 09 09 52](https://user-images.githubusercontent.com/1945542/69292019-e3a5fa80-0c3f-11ea-8b89-731d6e7ebbd1.png)

> I dug a little deeper into this. It looks like the StatsVersionStateCoordinator doesn't take wcAnalytics into account, only stats .v4 and .v4. (Could that be a factor?) I also logged a few things and noticed that the v4 endpoint was getting requested, despite the the wcAnalytics namespace defined in the remote request.

jaclync commented:
tested on @mindgraffiti 's store and the test cases worked, merging now - thanks all for testing!

jaclync commented:
@JavonDavis sorry for holding off the beta, it's now merged!

mindgraffiti commented:
Thank you @jaclync! I appreciate it.
